engine/Shopware/Components/Routing/Generators/DefaultGenerator.php

```
@@ -94,7 +86,7 @@ public function generate(array $params, Context $context)
 
         foreach ($params as $key => $value) {
             $route[] = $key;
-            $route[] = $value;
+            $route[] = is_array($value) || is_object($value) ? http_build_query($value) : $value;
```

Is there any specific reason why allowing objects? It feels wrong building that as query parameters

---

To allow using `\StdClass`. Otherwise, yes, no real reason besides that `http_build_query` allows objects to be passed ;)
I can remove it but then: what should happen with an object as this will silently fail again?

---

What about a trigger_error with an E_USER_DEPRECATED. And with 5.7 a exception.

---

Good point. I have added the triggering and modified the test accordingly. As this required replacing the _error_handler_ I had to move it to its own testfile to be able to clean up afterwards.